utility/servicer/module.go

```diff
@@ -138,19 +235,86 @@ func (s servicer) validateRelayChainSupport(relayChain *coreTypes.Identifiable,
 	return nil
 }
 
-// TODO: implement
 // validateApplication makes sure the application has not received more relays than allocated in the current session.
-func (s servicer) validateApplication(meta *coreTypes.RelayMeta, session *coreTypes.Session) error {
-	/*
-		// if maxRelaysPerSession, overServiced := calculateAppSessionTokens(); overServiced {
-			return fmt.Errorf("application %s has exceeded its allocated relays %d for the session %d", meta.ApplicationPublicKey, maxRelaysPerSession)
-		}
-	*/
-	return nil
+func (s *servicer) validateApplication(session *coreTypes.Session) error {
+	// IMPROVE: use a function to get current height from the current session
```

Comment:
1. Can't we just get it directly from `Session`?
2. If not, we should be able to do `s.GetBus().GetConsensusModule().CurrentHeight()`
3. If neither 1 or 2 work, this is `TECHDEBT`, not `IMPROVE` IMO

Reply:
Thank you, fixed: the called function now gets the height from the session.